tests/ga/test_extension.py

```diff
@@ -1464,24 +1464,32 @@ def test_uninstall_failure(self, patch_get_uninstall_command, *args):
         self._assert_no_handler_status(protocol.report_vm_status)
 
     @patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_update_command')
-    def test_upgrade_failure(self, patch_get_update_command, *args):
+    def test_extension_upgrade_failure_when_update_fails(self, patch_get_update_command, *args):
         """
-        Extension upgrade failure should not be retried
+        When the update command of the new extension fails, it should result in the new extension removed and the
+        old extension disabled. On the next goal state, the entire upgrade scenario should be retried (once),
+        meaning the new extension is downloaded again and update is called again.
         """
         test_data, exthandlers_handler, protocol = self._set_up_update_test_and_update_gs(patch_get_update_command,
                                                                                           *args)
 
-        exthandlers_handler.run()
-        self.assertEqual(1, patch_get_update_command.call_count)
+        with patch('azurelinuxagent.ga.exthandlers.HandlerManifest.get_enable_command') as patch_get_enable_command:
+            exthandlers_handler.run()
+            self.assertEqual(1, patch_get_update_command.call_count)
+            self.assertEqual(0, patch_get_enable_command.call_count)
 
-        # On the next iteration, update should not be retried
-        exthandlers_handler.run()
-        self.assertEqual(1, patch_get_update_command.call_count)
+            # On the next iteration, update should be retried.
+            exthandlers_handler.run()
+            self.assertEqual(1, patch_get_update_command.call_count)
+            self.assertEqual(0, patch_get_enable_command.call_count)
 
-        self._assert_handler_status(protocol.report_vm_status, "NotReady", expected_ext_count=1, version="1.0.1")
+        self.assertEquals("Ready", protocol.report_vm_status.call_args[0][0].vmAgent.status)
+
+        # Since the old extension is disabled, and the new one never got enabled, there is nothing to report.
+        self._assert_no_handler_status(protocol.report_vm_status)
```

Comment:
as discussed offline, the failure updating to the new version should be reported.

Reply:
Updating the tests to reflect this assertion earlier, after the first `exthandlers_handler.run()`, when the failure is reported.